iAM-legacy/Simulation/Investments.cs

```diff
@@ -369,14 +374,16 @@ public string IsBudgetAvailable(float fAmount, String strBudget, String strYear,
                     var currentYear = year + i;
 
                     //Priority must be all years, match the current year or the current year must be greater than maximum year and the Maximum year must be a priority.
-                    if(!(priority.IsAllYears || priority.Years == currentYear || (currentYear > MaximumYear && priority.Years==MaximumYear)))
+                    if (!(priority.IsAllYears || priority.Years == currentYear || (currentYear > MaximumYear && priority.Years == MaximumYear)))
```

Comment:
Priority domain logic belongs with priority objects and its data should be checked there, not in Investments.